### DISCUSSION TITLE
Want to host a server

### DISCUSSION THREAD

**Discussion by thewebcoder2009** [vercel/community#2243] [Help]
### Summary

I have hosted my server in vercel which has endpoints /api/auth and /api/notes for which I have defined two files in api folder which are auth.js and notes.js but when fetching data from endpoints its showing me FUNCTION_INVOCATION_FAILED.

Error:-

Unhandled Promise Rejection 	{"errorType":"Runtime.UnhandledPromiseRejection","errorMessage":"TypeError: Cannot read properties of undefined (reading 'apply')","reason":{"errorType":"TypeError","errorMessage":"Cannot read properties of undefined (reading 'apply')","stack":["TypeError: Cannot read properties of undefined (reading 'apply')","    at /var/task/node_modules/express/lib/router/index.js:646:15","    at next (/var/task/node_modules/express/lib/router/index.js:265:14)","    at Function.handle (/var/task/node_modules/express/lib/router/index.js:175:3)","    at router (/var/task/node_modules/express/lib/router/index.js:47:12)","    at /var/task/___vc/__helpers.js:476:20","    at AsyncLocalStorage.run (node:async_hooks:341:14)","    at Server.handler (/var/task/___vc/__launcher.js:3804:30)","    at Server.emit (node:events:513:28)","    at parserOnIncoming (node:_http_server:1072:12)","    at HTTPParser.parserOnHeadersComplete (node:_http_common:119:17)"]},"promise":{},"stack":["Runtime.UnhandledPromiseRejection: TypeError: Cannot read properties of undefined (reading 'apply')","    at process.<anonymous> (file:///var/runtime/index.mjs:1189:17)","    at process.emit (node:events:525:35)","    at emit (node:internal/process/promises:149:20)","    at processPromiseRejections (node:internal/process/promises:283:27)","    at process.processTicksAndRejections (node:internal/process/task_queues:96:32)"]}
Unknown application error occurred
Runtime.Unknown

**Comment by amyegan** [comment]
Hi @thewebcoder2009! This appears to be an uncaught exception, but I don't have the full context to know exactly what's causing it. 🤔 One common problem is that Express apps need to be set up a little differently from normal to work in a serverless environment. You can learn more about that in our guide on using Express.js with Vercel: https://vercel.com/guides/using-express-with-vercel
